src/c/tests/TestClient.cc

```diff
@@ -1,1397 +0,0 @@
-        zhandle_t *zk = createClient(&ctx);
-        int rc = zoo_create(zk, "/testarwg", "", 0, &ZOO_OPEN_ACL_UNSAFE, 0, 0, 0);
-        CPPUNIT_ASSERT_EQUAL((int) ZOK, rc);
-        rc = zoo_create(zk, "/testarwg/arwg", "", 0, &ZOO_OPEN_ACL_UNSAFE, 0, 0, 0);
-        CPPUNIT_ASSERT_EQUAL((int) ZOK, rc);
-      }
-
-      {
-        watchctx_t ctx;
-        zhandle_t *zk = createchClient(&ctx, "127.0.0.1:22181/testarwg/arwg");
-
-        testWatcherAutoReset(zk, &ctx, &ctx);
-      }
-    }
-
-    void testWatcherAutoResetWithLocal()
-    {
-      {
-        watchctx_t ctx;
-        zhandle_t *zk = createClient(&ctx);
-        int rc = zoo_create(zk, "/testarwl", "", 0, &ZOO_OPEN_ACL_UNSAFE, 0, 0, 0);
-        CPPUNIT_ASSERT_EQUAL((int) ZOK, rc);
-        rc = zoo_create(zk, "/testarwl/arwl", "", 0, &ZOO_OPEN_ACL_UNSAFE, 0, 0, 0);
-        CPPUNIT_ASSERT_EQUAL((int) ZOK, rc);
-      }
-
-      {
-        watchctx_t ctx;
-        watchctx_t lctx;
-        zhandle_t *zk = createchClient(&ctx, "127.0.0.1:22181/testarwl/arwl");
-        testWatcherAutoReset(zk, &ctx, &lctx);
-      }
-    }
-
-    void testLastZxid() {
-      // ZOOKEEPER-1417: Test that c-client only update last zxid upon
-      // receiving request response only.
-      const int timeout = 5000;
-      int rc;
-      watchctx_t ctx1, ctx2;
-      zhandle_t *zk1 = createClient(&ctx1);
-      zhandle_t *zk2 = createClient(&ctx2);
-      CPPUNIT_ASSERT(zk1);
-      CPPUNIT_ASSERT(zk2);
-
-      int64_t original = zk2->last_zxid;
-
-      // Create txn to increase system zxid
-      rc = zoo_create(zk1, "/lastzxid", "", 0,
-                      &ZOO_OPEN_ACL_UNSAFE, 0, 0, 0);
-      CPPUNIT_ASSERT_EQUAL((int)ZOK, rc);
-
-      // This should be enough time for zk2 to receive ping request
-      usleep(timeout/2 * 1000);
-
-      // check that zk1's last zxid is updated
-      struct Stat stat;
-      rc = zoo_exists(zk1, "/lastzxid", 0, &stat);
-      CPPUNIT_ASSERT_EQUAL((int) ZOK, rc);
-      CPPUNIT_ASSERT_EQUAL((int64_t) zk1->last_zxid, stat.czxid);
-      // zk2's last zxid should remain the same
-      CPPUNIT_ASSERT_EQUAL(original, (int64_t) zk2->last_zxid);
-
-      // Perform read and also register a watch
-      rc = zoo_wexists(zk2, "/lastzxid", watcher, &ctx2, &stat);
-      CPPUNIT_ASSERT_EQUAL((int) ZOK, rc);
-      int64_t updated = zk2->last_zxid;
-      // check that zk2's last zxid is updated
-      CPPUNIT_ASSERT_EQUAL(updated, stat.czxid);
-
-      // Increment system zxid again
-      rc = zoo_set(zk1, "/lastzxid", NULL, -1, -1);
-      CPPUNIT_ASSERT_EQUAL((int) ZOK, rc);
-
-      // Wait for zk2 to get watch event
-      CPPUNIT_ASSERT(waitForEvent(zk2, &ctx2, 5));
-      // zk2's last zxid should remain the same
-      CPPUNIT_ASSERT_EQUAL(updated, (int64_t) zk2->last_zxid);
-    }
-
-    static void watcher_remove_watchers(zhandle_t *zh, int type,
-                                    int state, const char *path,void *watcherCtx) {
-        count++;
```

Comment:
### ✅ No longer an issue: Excess Number of Function Arguments
watcher_remove_watchers is no longer above the threshold for number of arguments